tests/02_eradiate/01_unit/scenes/illumination/test_spot.py

```diff
@@ -0,0 +1,49 @@
+import numpy as np
+import pinttr
+import pytest
+
+from eradiate import unit_registry as ureg
+from eradiate.contexts import KernelDictContext
+from eradiate.scenes.illumination import SpotIllumination
+
+
+def test_construct_basic(mode_mono):
+    # We need a default spectral config
+    ctx = KernelDictContext()
+
+    # Construct without parameters
+    d = SpotIllumination()
+    assert d.kernel_dict(ctx).load() is not None
+
+    # Define all parameters
+    d = SpotIllumination(
+        target=[0, 0, 0],
+        origin=[1, 1, 1],
+        up=[1, 0, 1],
+        beam_width=10*ureg.deg,
+        intensity=10
+    )
+    assert d.kernel_dict(ctx).load() is not None
+
+
+def test_construct_from_size(mode_mono):
+    # We need a default spectral config
+    ctx = KernelDictContext()
+
+    spot_radius = 1 * ureg.m
+    beam_angle = 3 * ureg.deg
+
+    # spot_radius / tan(beam_angle/2.)
+    expected_distance = 38.18846
+
+    d = SpotIllumination.from_size(
+        target=[0, 0, 0],
+        direction=[0, 0, -1],
+        spot_radius=spot_radius,
+        beam_divergence=beam_angle
+    )
+
+    assert np.allclose(d.origin.m_as(ureg.m), [0, 0, expected_distance])
+
+    assert d.kernel_dict(ctx).load() is not None
+
```

Comment:
Perhaps, we could add a test where the `beam_profile` parameter is set. Since it is a file path, I guess things can easily go wrong there, so it'd be nice to confirm that this is working.

Reply:
Good point, but then I need a fixture that creates a temporary bitmap file :D